### PR TITLE
[Dist] enable access DistGraph.edges via canonical etype

### PR DESCRIPTION
**Pull request by Rhett-Ying**
## Description

we should always use canonical etypes as canonical etype is fully supported in DistDGL.
## Checklist
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [ ] I've leverage the [tools](https://docs.google.com/document/d/1iHyj7zlmygKSk5gBPsqIqL5ASPzJSPREaNT_QdsiYA4/edit) to beautify the python and c++ code.
- [ ] The PR is complete and small, read the [Google eng practice (CL equals to PR)](https://google.github.io/eng-practices/review/developer/small-cls.html) to understand more about small PR. In DGL, we consider PRs with less than 200 lines of core code change are small (example, test and documentation could be exempted).
- [ ] All changes have test coverage
- [ ] Code is well-documented
- [ ] To the best of my knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change
- [ ] Related issue is referred in this PR
- [ ] If the PR is for a new model/paper, I've updated the example index [here](../examples/README.md).

### CODE REVIEW

File: tests/distributed/test_dist_graph_store.py [hunk 1]
```diff
@@ -730,7 +730,8 @@ def check_dist_graph_hetero(g, num_clients, num_nodes, num_edges):
 
     # Test reading edge data
     eids = F.arange(0, int(g.number_of_edges("r1") / 2))
-    feats1 = g.edges["r1"].data["feat"][eids]
+    c_etype = g.to_canonical_etype("r1")
+    feats1 = g.edges[c_etype].data["feat"][eids]
```
Comment (jermainewang) [inline tests/distributed/test_dist_graph_store.py:734]: Is this PR a breaking change? If not, we shouldn't change the unit test but add a new test instead.

File: python/dgl/distributed/dist_graph.py [hunk 2]
```diff
@@ -157,14 +157,17 @@ def __getitem__(self, key):
...
-    """A NodeView class to act as G.nodes for a DistGraph."""
+    """A EdgeView class to act as G.edges for a DistGraph."""
     __slots__ = ['_graph']
 
     def __init__(self, graph):
         self._graph = graph
 
     def __getitem__(self, key):
-        assert isinstance(key, str)
+        assert isinstance(key, str) or \
```
Comment (frozenbugs) [inline python/dgl/distributed/dist_graph.py:167]:
```suggestion
        assert isinstance(key, str) or ( 
            isinstance(key, tuple) and len(key) == 3 
        ), f"Expect edge type in string or a triplet of string, but got {key}."
```

Comment (frozenbugs) [reply python/dgl/distributed/dist_graph.py:167]: Let's avoid using \ to change newline.
Other smarter way to increase readability is to add local variable for "isinstance(key, tuple) and len(key) == 3 ".

File: python/dgl/distributed/graph_services.py [hunk 3]
```diff
@@ -638,9 +637,18 @@ def sample_etype_neighbors(
...
-        fanout = F.full_1d(len(g.etypes), fanout, F.int64, F.cpu())
+        fanout = F.full_1d(len(g.canonical_etypes), fanout, F.int64, F.cpu())
     else:
-        fanout = F.tensor([fanout[etype] for etype in g.etypes], dtype=F.int64)
+        etype_ids = {etype: i for i, etype in enumerate(g.canonical_etypes)}
+        fanout_ = [None] * len(g.canonical_etypes)
+        for etype, v in fanout.items():
+            c_etype = g.to_canonical_etype(etype)
+            fanout_[etype_ids[c_etype]] = v
+        assert all(v is not None for v in fanout_), \
```
Comment (frozenbugs) [inline python/dgl/distributed/graph_services.py:647]: ditto.

File: tests/distributed/test_dist_graph_store.py (hunk 1 above)
Comment (Rhett-Ying) [reply tests/distributed/test_dist_graph_store.py:734]: It's not a breaking change. The change I made for existing test is add more tests to cover the change here, not identical to modifying existing one. In specific here, access via `"r1"` is tested in other part in this testcase, so I think it's ok to change the access way here to cover the access via canonical etype.

**Review by frozenbugs — APPROVED** [review]
Approve for the good readability.

File: python/dgl/distributed/dist_graph.py [hunk 4]
```diff
@@ -157,14 +157,16 @@ def __getitem__(self, key):
...
+    """A EdgeView class to act as G.edges for a DistGraph."""
     __slots__ = ['_graph']
 
     def __init__(self, graph):
         self._graph = graph
 
     def __getitem__(self, key):
-        assert isinstance(key, str)
+        assert isinstance(key, str) or (
+            isinstance(key, tuple) and len(key) == 3
```
Comment (Kh4L) [inline python/dgl/distributed/dist_graph.py:168]: How about 
```suggestion
            isinstance(key, (tuple, list)) and len(key) == 3
```

Comment (Rhett-Ying) [reply python/dgl/distributed/dist_graph.py:168]: canonical etype is always in tuple, should not be a list.

Comment (Kh4L) [reply python/dgl/distributed/dist_graph.py:168]: At first I thought that there could be cases there they might be created by the user, but that makes sense

File: tests/distributed/test_dist_graph_store.py (hunk 1 above)
Comment (Rhett-Ying) [reply tests/distributed/test_dist_graph_store.py]: accessing via etype and canonical etype are both added.

File: python/dgl/distributed/graph_services.py [hunk 5]
```diff
@@ -638,9 +637,19 @@ def sample_etype_neighbors(
         A sampled subgraph containing only the sampled neighboring edges.  It is on CPU.
     """
     if isinstance(fanout, int):
-        fanout = F.full_1d(len(g.etypes), fanout, F.int64, F.cpu())
+        fanout = F.full_1d(len(g.canonical_etypes), fanout, F.int64, F.cpu())
     else:
-        fanout = F.tensor([fanout[etype] for etype in g.etypes], dtype=F.int64)
+        etype_ids = {etype: i for i, etype in enumerate(g.canonical_etypes)}
+        fanout_ = [None] * len(g.canonical_etypes)
```
Comment (jermainewang) [inline python/dgl/distributed/graph_services.py:643]: fanout_ -> fanout_array  . The name is too similar to `fanout` otherwise.

**Pull request merged**